### DISCUSSION TITLE
Token warning (if something is about to exceed the prompt context window and cause the task to fail out) / Third Option to edit the previous message and remove the problematic folder/file

### DISCUSSION THREAD

**Discussion by arimxyer** [cline/cline#731] [Feature Requests]
Would be helpful if there was a warning message ahead of sending a message during a successful task thread if adding in a folder or file is about to error out the current thread. If this happens, it only leaves the two options of "Retry" and "Start a new task" - would also be helpful if there was a third option to edit the previous message, so you can continue on with the same task without losing all your previous history and context.

**Comment by farbour** [comment]
+1
